php/includes/language.php

```diff
@@ -91,3 +91,40 @@ function FN_(string $singular, string $plural, ...$replacePairs)
 		wrap_array_keys_in_braces($replacePairs),
 	);
 }
+
+// custom translation framework
+define(
+	"TRANSLATION_TABLE",
+	array(
+		"kHTML_LANGUAGE" => array(
```

Comment:
Very simple structure: `string_key` −> `language` -> `translated string`. Not sure how we could add context in there. It would either be a 3rd layer or somehow using mangling.

Reply:
I think we can take inspiration from `i18n` they already gave some solutions for that

Reply:
For example this is how they handle the plural: https://www.i18next.com/translation-function/plurals

Reply:
I also noticed that they inverted the structure: language -> string_key -> translated string, I think it's cleaner this way as you don't have to repeat 'en' => aaa, 'fr' => bbb every time

Reply:
Discussed offline, we're going to keep `key` -> `language` -> `translation` as it keeps sentences that have a close context together, which helps for translation.